lib/facter/custom_facts/util/fact.rb

```diff
@@ -85,7 +85,7 @@ def define_resolution(resolution_name, options = {}, &block)
 
         resolve
       rescue StandardError => e
-        log.log_exception("Unable to add resolve #{resolution_name.inspect} for fact #{@name}: #{e.message}")
+        log.log_exception(e)
```

Comment:
I'm wondering if the exception will always contain enough information to troubleshoot which fact name and ideally file location needs to be fixed? If not, may be we keep the message and just pass the exception as the first argument like: 
```ruby
log_exception(e, "Unable to add resolve #{resolution_name.inspect} for fact #{@name}: #{e.message}")
```

Reply:
I tried that at first, but this method only takes an exception as an argument and gave me an ArgumentError when I tried to pass the message as well: https://github.com/puppetlabs/facter/blob/163e1bb6b7440d62c8326cd7e916f2394aad3ab2/lib/facter/framework/logging/logger.rb#L163